### ISSUE TITLE
Issue Finding odbc PostgreSQL Driver

### ISSUE THREAD

**Issue by wbakerrobinson** [r-dbi/odbc#352]
### Issue Description and Expected Result
I am really struggling to find my odbc driver for Postgres. I have been trying now for 5+ hours. Had a TA help me, etc. Hopefully someone here will be able to give me an idea of how odbc searches for my driver, and what I need to do to find it.

Code where I try to connect:
server <- function(input, output) {
    output$report <- downloadHandler(
        filename = "cs586Project.html",
        content = function(file){
            tempReport <- file.path(tempdir(), "cs586Project.Rmd")
            file.copy("cs586Project.Rmd", tempReport, overwrite = TRUE)
            #parameters passed to .rmd
            params <- list(availSurv = readCSVparam(input$availSurv), 
                           projSurv = readCSVparam(input$projSurv), 
                           eligible = readXLparam(input$eligibile),
                           projInfo = readCSVparam(input$projInfo), 
                           conn = dbConnect(odbc::odbc(),
                               driver = "PostgreSQL Driver",
                               dbname = input$dbname,
                               host = input$host,
                               username = input$username,
                               password = input$dbPassword,
                               port = 5432
                           ))
            rmarkdown::render(tempReport, output_file = file,
                              params = params,
                              envir = new.env(parent = globalenv()))
        }
    )
}

> runApp('cs586DBproject')
Listening on http://127.0.0.1:7055
Warning: Error in : nanodbc/nanodbc.cpp:983: 00000: [unixODBC][Driver Manager]Can't open lib 'PostgreSQL Driver' : file not found 
  [No stack trace available]

What my odbcinst.ini looks like:
![Screen Shot 2020-03-10 at 14 30 33](https://user-images.githubusercontent.com/49573405/76361208-be66f600-62db-11ea-9ec2-f5d4a5343728.png)

Where my odbscinst.ini is stored:
![Screen Shot 2020-03-10 at 14 24 55](https://user-images.githubusercontent.com/49573405/76360787-f0c42380-62da-11ea-8503-31845cabfa52.png)

### Database
PostgreSQL 9.5

### Reproducible Example
> odbc::odbcListDrivers()
[1] name      attribute value    
<0 rows> (or 0-length row.names)

From reprex:
Removing leading prompts from reprex source.
Rendering reprex...
 
Error: <callr_status_error: callr subprocess failed: <text>:19:1: unexpected '['
18: odbc::odbcListDrivers()
19: [
    ^>
-->
<callr_remote_error in parse(text = x, keep.source = TRUE):
 <text>:19:1: unexpected '['
18: odbc::odbcListDrivers()
19: [
    ^>
 in process 3989 

See `.Last.error.trace` for a stack trace.

─ Session info ──────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.1 (2019-07-05)
 os       macOS Mojave 10.14.6        
 system   x86_64, darwin15.6.0        
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/Los_Angeles         
 date     2020-03-10                  

─ Packages ──────────────────────────────────────────────────────────────────────────────────
 package     * version    date       lib source                       
 assertthat    0.2.1      2019-03-21 [1] CRAN (R 3.6.0)               
 backports     1.1.5      2019-10-02 [1] CRAN (R 3.6.0)               
 bit           1.1-15.2   2020-02-10 [1] CRAN (R 3.6.0)               
 bit64         0.9-7      2017-05-08 [1] CRAN (R 3.6.0)               
 blob          1.2.1      2020-01-20 [1] CRAN (R 3.6.0)               
 callr         3.4.0      2019-12-09 [1] CRAN (R 3.6.0)               
 cellranger    1.1.0      2016-07-27 [1] CRAN (R 3.6.0)               
 cli           2.0.2      2020-02-28 [1] CRAN (R 3.6.0)               
 clipr         0.7.0      2019-07-23 [1] CRAN (R 3.6.0)               
 crayon        1.3.4      2017-09-16 [1] CRAN (R 3.6.0)               
 DBI         * 1.1.0      2019-12-15 [1] CRAN (R 3.6.0)               
 dbplyr        1.4.2      2019-06-17 [1] CRAN (R 3.6.0)               
 desc          1.2.0      2018-05-01 [1] CRAN (R 3.6.0)               
 devtools    * 2.2.1      2019-09-24 [1] CRAN (R 3.6.0)               
 digest        0.6.25     2020-02-23 [1] CRAN (R 3.6.0)               
 dplyr         0.8.4      2020-01-31 [1] CRAN (R 3.6.0)               
 ellipsis      0.3.0      2019-09-20 [1] CRAN (R 3.6.0)               
 evaluate      0.14       2019-05-28 [1] CRAN (R 3.6.0)               
 fansi         0.4.1      2020-01-08 [1] CRAN (R 3.6.0)               
 fastmap       1.0.1      2019-10-08 [1] CRAN (R 3.6.0)               
 fs            1.3.1      2019-05-06 [1] CRAN (R 3.6.0)               
 glue          1.3.1      2019-03-12 [1] CRAN (R 3.6.0)               
 hms           0.5.3      2020-01-08 [1] CRAN (R 3.6.0)               
 htmltools     0.4.0      2019-10-04 [1] CRAN (R 3.6.0)               
 httpuv        1.5.2      2019-09-11 [1] CRAN (R 3.6.0)               
 jsonlite      1.6        2018-12-07 [1] CRAN (R 3.6.0)               
 knitr         1.27       2020-01-16 [1] CRAN (R 3.6.1)               
 later         1.0.0      2019-10-04 [1] CRAN (R 3.6.0)               
 magrittr      1.5        2014-11-22 [1] CRAN (R 3.6.0)               
 memoise       1.1.0      2017-04-21 [1] CRAN (R 3.6.0)               
 mime          0.8        2019-12-19 [1] CRAN (R 3.6.0)               
 odbc        * 1.2.2      2020-01-10 [1] CRAN (R 3.6.0)               
 packrat       0.5.0      2018-11-14 [1] CRAN (R 3.6.0)               
 pillar        1.4.3      2019-12-20 [1] CRAN (R 3.6.0)               
 pkgbuild      1.0.6      2019-10-09 [1] CRAN (R 3.6.0)               
 pkgconfig     2.0.3      2019-09-22 [1] CRAN (R 3.6.0)               
 pkgload       1.0.2      2018-10-29 [1] CRAN (R 3.6.0)               
 pool        * 0.1.4.9000 2020-03-08 [1] Github (rstudio/pool@0b03680)
 prettyunits   1.1.0      2020-01-09 [1] CRAN (R 3.6.0)               
 processx      3.4.1      2019-07-18 [1] CRAN (R 3.6.0)               
 promises      1.1.0      2019-10-04 [1] CRAN (R 3.6.0)               
 ps            1.3.0      2018-12-21 [1] CRAN (R 3.6.0)               
 purrr         0.3.3      2019-10-18 [1] CRAN (R 3.6.0)               
 R6            2.4.1      2019-11-12 [1] CRAN (R 3.6.0)               
 Rcpp          1.0.3      2019-11-08 [1] CRAN (R 3.6.0)               
 readxl      * 1.3.1      2019-03-13 [1] CRAN (R 3.6.0)               
 remotes       2.1.0      2019-06-24 [1] CRAN (R 3.6.0)               
 reprex        0.3.0      2019-05-16 [1] CRAN (R 3.6.0)               
 rlang         0.4.5      2020-03-01 [1] CRAN (R 3.6.0)               
 rmarkdown     2.0        2019-12-12 [1] CRAN (R 3.6.0)               
 RPostgres   * 1.2.0      2019-12-18 [1] CRAN (R 3.6.0)               
 rprojroot     1.3-2      2018-01-03 [1] CRAN (R 3.6.0)               
 rsconnect     0.8.16     2019-12-13 [1] CRAN (R 3.6.1)               
 rstudioapi    0.10       2019-03-19 [1] CRAN (R 3.6.0)               
 sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.6.0)               
 shiny       * 1.4.0      2019-10-10 [1] CRAN (R 3.6.0)               
 sourcetools   0.1.7      2018-04-25 [1] CRAN (R 3.6.0)               
 testthat      2.3.1      2019-12-01 [1] CRAN (R 3.6.0)               
 tibble        2.1.3      2019-06-06 [1] CRAN (R 3.6.0)               
 tidyselect    1.0.0      2020-01-27 [1] CRAN (R 3.6.0)               
 usethis     * 1.5.1      2019-07-04 [1] CRAN (R 3.6.0)               
 vctrs         0.2.3      2020-02-20 [1] CRAN (R 3.6.0)               
 whisker       0.4        2019-08-28 [1] CRAN (R 3.6.0)               
 withr         2.1.2      2018-03-15 [1] CRAN (R 3.6.0)               
 xfun          0.12       2020-01-13 [1] CRAN (R 3.6.0)               
 xtable        1.8-4      2019-04-21 [1] CRAN (R 3.6.0)               

[1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library

**Comment by jimhester** [comment]
This sort of question is a better fit for <https://community.rstudio.com>. Do you mind asking it over there? (You might want to read <https://www.tidyverse.org/help/> first to maximise your chances of getting a good answer)

**Comment by wbakerrobinson** [comment]
I did pose this question on community.rstudio.com (https://community.rstudio.com/t/postgres-sql-db-connection-with-dbi-ocdb/56072). After spending 20 hours trying to get things working, thoroughly reading tidyverse's help, and all obdc related material I could find, I ended up having to use reticulate and psycopg2. If it is not a bug with your package it seems to be a major flaw with the way it is implemented. Should not be that hard to get this obdc driver working. My obdc postgres driver works outside R from the terminal, and in other applications just not with your package.
